### PR TITLE
Prepare for the release v0.9.0

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR does various changes in order to prepare for the release v0.9.0. The release is targeted for Monday, July 1st.

* Make the installation steps in the `README.md` file KubeOne version-agnostic
* Update the `README.md` file to recommend using officially released versions over the `master` branch
* Mention that the support for package managers is work in progress
* Add Kubernetes 1.15 to the compatibility table
* Update the release process document
* Add changelog for the v0.9.0 release

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by xmudrii** [comment]
We're release blocking clear. Rebased the PR. I'm starting tests on DigitalOcean to ensure the status.

/test pull-kubeone-e2e-digitalocean-conformance-1.14
/test pull-kubeone-e2e-digitalocean-conformance-1.15
/test pull-kubeone-e2e-digitalocean-upgrade-1.13-1.14
/test pull-kubeone-e2e-digitalocean-upgrade-1.13-1.14

**Comment by xmudrii** [comment]
@kron4eg I added everything merged recently to the CHANGELOG. PTAL

File: README.md [hunk 1]
```diff
@@ -94,12 +137,12 @@ In the following table you can find what are supported Kubernetes and Terraform
...
-| KubeOne version | 1.14 | 1.13 | Terraform | Supported providers                                                |
-|-----------------|------|------|-----------|--------------------------------------------------------------------|
-| v0.9.0+         | +    | +    | v0.12+    | AWS, DigitalOcean, GCE, Hetzner, Packet, OpenStack, vSphere, Azure |
-| v0.8.0+         | +    | +    | v0.11     | AWS, DigitalOcean, GCE, Hetzner, Packet, OpenStack, vSphere        |
-| v0.6.0+         | +    | +    | v0.11     | AWS, DigitalOcean, GCE, Hetzner, Packet, OpenStack                 |
-| v0.5.0          | +    | +    | v0.11     | AWS, DigitalOcean, GCE, Hetzner, OpenStack                         |
+| KubeOne version | 1.15 | 1.14 | 1.13 | Terraform | Supported providers                                                |
+|-----------------|------|------|------|-----------|--------------------------------------------------------------------|
+| v0.9.0+         | +    | +    | +    | v0.12+    | AWS, DigitalOcean, GCE, Hetzner, Packet, OpenStack, vSphere, Azure |
+| v0.8.0+         | -    | +    | +    | v0.11     | AWS, DigitalOcean, GCE, Hetzner, Packet, OpenStack, vSphere        |
```
Comment (kron4eg) [inline README.md]: I think it's possible to install 1.15 using v0.8 version... and probably even previous version as well.

Comment (xmudrii) [reply README.md]: It is but I'd not recommend it considering we had some needed changes that are not backported. What comes to my mind is the default admission controllers.

**Comment by xmudrii** [comment]
Closing and reopening to ensure something.

**Comment by xmudrii** [comment]
/hold cancel
:tada: :shipit:

**Pull request merged**